test/integration/otel_test.go

```diff
@@ -0,0 +1,287 @@
+				rpcSpan("ra.RegistrationAuthority/PerformValidation", wfe, ra,
+					rpcSpan("sa.StorageAuthority/GetRegistration", ra, sa))),
+			httpSpan("/acme/finalize/",
+				rpcSpan("sa.StorageAuthorityReadOnly/GetOrder", wfe, sa),
+				rpcSpan("ra.RegistrationAuthority/FinalizeOrder", wfe, ra,
+					rpcSpan("sa.StorageAuthority/KeyBlocked", ra, sa),
+					rpcSpan("sa.StorageAuthority/GetRegistration", ra, sa),
+					rpcSpan("sa.StorageAuthority/GetValidOrderAuthorizations2", ra, sa),
+					rpcSpan("sa.StorageAuthority/SetOrderProcessing", ra, sa),
+					rpcSpan("ca.CertificateAuthority/IssuePrecertificate", ra, ca),
+					rpcSpan("Publisher/SubmitToSingleCTWithResult", ra, "boulder-publisher"),
+					rpcSpan("ca.CertificateAuthority/IssueCertificateForPrecertificate", ra, ca),
+					rpcSpan("sa.StorageAuthority/FinalizeOrder", ra, sa))),
+			httpSpan("/acme/order/", rpcSpan("sa.StorageAuthorityReadOnly/GetOrder", wfe, sa)),
+			httpSpan("/acme/cert/", rpcSpan("sa.StorageAuthorityReadOnly/GetCertificate", wfe, sa)),
+		},
+	}
+
+	// Retry checking for spans. Span submission is batched asynchronously, so we
+	// may have to wait for the DefaultScheduleDelay (5 seconds) for results to
+	// be available. Rather than always waiting, we retry a few times.
+	// Empirically, this test passes on the second or third try.
+	var traceData TraceData
+	found := false
+	const retries = 10
+	for try := 0; try < retries; try++ {
+		traceData := getTraceFromJaeger(t, traceID)
+		if findSpans(traceData, "", expectedSpans) {
+			found = true
+			break
+		}
+		time.Sleep(sdktrace.DefaultScheduleDelay / 5 * time.Millisecond)
+	}
+	test.Assert(t, found, fmt.Sprintf("Failed to find expected spans in Jaeger for trace %s", traceID))
+
+	test.AssertEquals(t, len(traceData.Warnings), 0)
+	for _, span := range traceData.Spans {
+		for _, warning := range span.Warnings {
+			if strings.Contains(warning, "clock skew adjustment disabled; not applying calculated delta") {
+				continue
+			}
+			t.Errorf("Span %s (%s) warning: %v", span.SpanID, span.OperationName, warning)
+		}
+	}
+}
+
+func traceIssuingTestCert(t *testing.T) trace.TraceID {
+	domains := []string{random_domain()}
+
+	// Configure this integration test to trace to jaeger:4317 like Boulder will
+	shutdown := cmd.NewOpenTelemetry(cmd.OpenTelemetryConfig{
+		Endpoint:    "jaeger:4317",
+		SampleRatio: 1,
+	}, blog.Get())
+	defer shutdown(context.Background())
+
+	tracer := otel.GetTracerProvider().Tracer("TraceTest")
+	ctx, span := tracer.Start(context.Background(), "TraceTest")
+	defer span.End()
+
+	// Provide an HTTP client with otel spans.
+	// The acme client doesn't pass contexts through, so we inject one.
+	option := acme.WithHTTPClient(&http.Client{
+		Timeout:   60 * time.Second,
+		Transport: &ContextInjectingRoundTripper{ctx},
+	})
+
+	c, err := acme.NewClient("http://boulder.service.consul:4001/directory", option)
+	test.AssertNotError(t, err, "newAccount failed")
```

Comment:
while looking at this test with @pgporada just now I noticed that I've copied the same message for a few asserts here, and that needs to be updated per call

Reply:
Good spotting